PocketKit/Tests/PocketKitTests/UrlValidatorTests.swift

```diff
@@ -0,0 +1,27 @@
+// This Source Code Form is subject to the terms of the Mozilla Public
+// License, v. 2.0. If a copy of the MPL was not distributed with this
+// file, You can obtain one at http://mozilla.org/MPL/2.0/.
+
+import XCTest
+
+@testable import PocketKit
+
+class UrlValidatorTests: XCTestCase { 
+    func testCorrectUrlReturnsItemUrl() {
+        let validator = UrlValidator()
+        let correctUrlString = "pocket:/app/openURL?url=https://example.com/this_is_an_article/"
+        let correctUrl = URL(string: correctUrlString)
+        XCTAssertNotNil(correctUrl)
+        let itemUrl = validator.getItemUrl(from: correctUrl!)
+        XCTAssertNotNil(itemUrl)
+        XCTAssertEqual(itemUrl, "https://example.com/this_is_an_article/")
+    }
+
+    func testInvalidUrlReturnsNil() {
+        let validator = UrlValidator()
+        let invalidUrl = URL(string: "https://getpocket.com/hooray/not-an-article-from-widgets")
+        XCTAssertNotNil(invalidUrl)
+        let itemUrl = validator.getItemUrl(from: invalidUrl!)
+        XCTAssertNil(itemUrl)
+    }
```

Comment:
Can you add one more test that throws some random garbage URL into the validator?

Reply:
That would be a test for `URL.init(_:)`, so I don't think it fits here.

Reply:
More what I was getting at was adding a valid URL that looks nothing like what the validator would be expecting

```let weirdoURL = URL(string: "testing:some.url.that.is.still.a/~valid/url?but=also.has&aBunch=of,stuff&array=that&array=we&array=dont&usually=have1234567890$-_.+!*'()")```